gpt/gpt/gpt.py

```diff
@@ -18,29 +19,34 @@ class Question(pc.Model, table=True):
 
     username: str
     prompt: str
-    question: str
+    answer: str
+    timestamp: datetime.datetime = datetime.datetime.now()
 
 
 class State(pc.State):
     """The app state."""
 
-    username: str
-    password: str
+    username: str = ""
+    password: str = ""
     logged_in: bool = False
 
-    prompt: str
-    result: str
+    prompt: str = ""
+    result: str = ""
 
     @pc.var
     def questions(self) -> list[Question]:
         """Get the users saved questions and answers from the database."""
         with pc.session() as session:
             if self.logged_in:
-                return (
+                qa = (
                     session.query(Question)
                     .where(Question.username == self.username)
+                    .distinct(Question.prompt)
```

Comment:
Should we also add an `order_by` on the `timestamp` so we get the questions in order?